lib/response_bank.rb

```diff
@@ -29,17 +30,30 @@ def read_from_backing_cache_store(_env, cache_key, backing_cache_store: cache_st
       backing_cache_store.read(cache_key, raw: true)
     end
 
-    def compress(content)
-      io = StringIO.new
-      gz = Zlib::GzipWriter.new(io)
-      gz.write(content)
-      io.string
-    ensure
-      gz.close
+    def compress(content, encoding = 'br')
+      case encoding
+      when 'gzip'
+        io = StringIO.new
+        gz = Zlib::GzipWriter.new(io)
+        gz.write(content)
+        gz.close
```

Comment:
We removed `gz.close` from the ensure block. We probably want to keep it there if we can.